src/main/java/cart/application/order/OrderCommandService.java

```diff
@@ -0,0 +1,115 @@
+package cart.application.order;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import cart.application.order.dto.OrderRequest;
+import cart.domain.cart.Cart;
+import cart.domain.cart.CartItemRepository;
+import cart.domain.cart.CartRepository;
+import cart.domain.coupon.MemberCoupon;
+import cart.domain.coupon.MemberCouponRepository;
+import cart.domain.coupon.type.CouponInfo;
+import cart.domain.coupon.type.NotUsed;
+import cart.domain.monetary.DeliveryFee;
+import cart.domain.order.Order;
+import cart.domain.order.OrderItem;
+import cart.domain.order.OrderRepository;
+import cart.domain.order.OrderStatus;
+import cart.domain.order.Orders;
+import cart.error.exception.BadRequestException;
+import cart.error.exception.ConflictException;
+import cart.error.exception.ForbiddenException;
+
+@Transactional
+@Service
+public class OrderCommandService {
+
+	private final OrderRepository orderRepository;
+	private final MemberCouponRepository memberCouponRepository;
+	private final CartRepository cartRepository;
+	private final CartItemRepository cartItemRepository;
+
+	public OrderCommandService(final OrderRepository orderRepository,
+		final MemberCouponRepository memberCouponRepository,
+		final CartRepository cartRepository, final CartItemRepository cartItemRepository) {
+		this.orderRepository = orderRepository;
+		this.memberCouponRepository = memberCouponRepository;
+		this.cartRepository = cartRepository;
+		this.cartItemRepository = cartItemRepository;
+	}
+
+	public Long addOrder(final Long memberId, final OrderRequest request) {
+		final Order order = createOrder(memberId, request);
+		validateTotalPriceOfOrder(request, order);
+
+		final Long savedId = orderRepository.save(memberId, order);
+		cartItemRepository.deleteByIds(request.getCartItemIds());
+
+		return savedId;
+	}
+
+	public void cancelOrder(final Long memberId, final Long orderId) {
+		final Orders orders = new Orders(memberId, orderRepository.findByMemberId(memberId));
+		final Order order = orders.getOrder(orderId)
+			.orElseThrow(ForbiddenException.Order::new);
```

Comment:
존재하지 않는 케이스에 대해 Forbidden 을 사용해주신 이유가 궁금해요. Forbidden은 어떤 의미로 사용되는 상태값일까요?

Reply:
<img width="540" alt="image" src="https://github.com/woowacourse/jwp-shopping-order/assets/112045553/65bb6f7c-3817-4fbb-9286-554c5f2d89f6">

안 그래도 크루들과 논의하고 있던 내용입니다.
일단 이번에는 협업 페어들과 Forbidden을 사용하기로 합의를 보아서 Forbidden으로 결정하였습니다.

기존 뼈대 코드에서는 아이템을 찾은 후, 아이템 안에 있는 멤버를 검증하여 일치하지 않으면 Forbidden을 발생시켰습니다.
저는 리팩터링을 했지만 리팩터링을 하지않은 백엔드 페어가 있어서
프론트엔드 크루들에게 같은 상황에서 같은 상태코드를 반환하기 위해 Forbidden을 사용했습니다.

조앤은 지금 제 코드 상태에선 Not Found를 사용하는 것이 더 적절한 사용이라고 판단하시나요?

Reply:
제 생각에 forbidden과 not found의 차이는 아래와 같아요.
* forbidden: 클라이언트의 access가 허용되지 않음. 존재하더라도 클라이언트에게 permission 자체가 없어 보여줄 수 없는 경우
* not found: 요청한 리소스를 리턴해야하는데 해당 리소스가 존재하지 않을 때

말씀해주신 케이스에서 다른 유저의 장바구니 아이템을 요청한 경우에는 forbidden이 맞는 것 같고, 해당 아이템이 존재하지 않는 경우는 not found가 맞는 것 같아요!ㅎㅎ 케이스를 구분하는게 좀 더 명확해 보이는데, 404는 403까지 범용적으로 커버할 수 있다고 하네요 (의미상?)

> The 404 status code can also be used in 403 scenarios, when the server does not want to send back the reason why it is refusing to serve the request. A good example is when the server senses some kind of an attack, which might be a brute force attack. In this case, the server responds with a 404 Not found instead of a 403 Forbidden and an explanation.